backend/trojan_dining/views.py

```diff
@@ -81,7 +83,34 @@ def post(self, request):
                 return HttpResponse(400)
             
             try:  
-                Subscription(item_id = item_id, email = email, phone_no = phone_number).save()
+                try:
+                    Subscription.objects.get(item_id = item_id, email = email, phone_no = phone_number)
+                except:
+                    Subscription(item_id = item_id, email = email, phone_no = phone_number).save()
                 return HttpResponse(201)
             except:
                 return HttpResponse(500)
+
+class GetSubscriptions(APIView):
+    def get(self, request):
+        user_email = request.GET.get('email')
+
+        if user_email is None:
+            return HttpResponse(status = 400)
+        else:
+            subscription_list = list(Subscription.objects.filter(email = user_email).values())
```

Comment:
Does this need to be cast to a list? Fine if it does, but I thought we could loop through this queryset

Reply:
Could you elaborate on what you mean by this?

Reply:
My (not great) understanding is that when you cast to a list like this, it casts it from a queryset object to something closer to an actual model object. Again I don't have a great understanding of this, I was just wondering if we could loop through this without casting to a list. I'm fine with merging this too 👍